### PR TITLE
Bump `sidekiq` to `3.5.4`

### PR DESCRIPTION
**Pull request by bishboria**
_Do not merge straight away_

Ensure `sidekiq` has no work to retry, see https://github.com/mperham/sidekiq/blob/master/3.0-Upgrade.md

Part of: https://trello.com/c/1mXjOji6/565-upgrade-rummager-dependencies

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ gem "logging", "2.1.0"
 gem 'nokogiri', "1.6.7.2"
 gem 'whenever', require: false
 gem "slop", "3.4.5"
-gem "sidekiq", "< 3"
+gem "sidekiq", "< 4"
```
Comment (Davidslv) [inline Gemfile]: shouldn't we lock it to `3.4.1` instead? what's the chances of someone upgrading to 3.5.4 ?

Comment (bishboria) [inline Gemfile]: When this PR goes in, another should be raised to upgrade `sidekiq`:
- https://github.com/mperham/sidekiq/blob/master/3.0-Upgrade.md
- https://github.com/mperham/sidekiq/blob/master/4.0-Upgrade.md

The `< X` is recommended in those upgrade documents too. I think when the last sidekiq bump goes in, it should lock the version

**Comment by tijmenb** [comment]
This can't be merged before we've deployed the previous version to production.

**Comment by tijmenb** [comment]
Why `3.4.1` and not `3.5.4`?

**Comment by bishboria** [comment]
Where does this version `3.5.4` come from? Sidekiq doesn't have that version.

**Comment by tijmenb** [comment]
https://rubygems.org/gems/sidekiq/versions

**Comment by bishboria** [comment]
Bundler picks version `3.4.1` itself. Manually setting `sidekiq` to `3.5.4` gets a compatibility error with `redis-namespace`.

**Comment by tijmenb** [comment]
So why is `redis-namespace` still locked?

**Comment by bishboria** [comment]
I was only looking at sidekiq just now :)

**Comment by tijmenb** [comment]
`redis-namespace` is only in there for sidekiq, we don't use it for something else if I'm not mistaking. We could just remove it and let sidekiq handle that dependency.

**Comment by bishboria** [comment]
Yeah, that's what I'm doing now. Making sure it's all running on the dev vm first.

**Comment by h-lame** [comment]
Note that sidekiq 4+ drops the redis-namespace dependency, so if we are using it for sidekiq it might be better to leave it in as an explicit dep to avoid breakages when you go up to sidekiq 4.

**Comment by tijmenb** [comment]
Ah, thanks @h-lame! In that case, setting `"redis-namespace", "~> 1.5.2"` would make sense.

**Comment by tijmenb** [comment]
This is now mergable because we deployed Rummager to production on Friday.

**Pull request merged**